odmantic/exceptions.py

```diff
@@ -43,7 +44,11 @@ class DuplicateKeyError(BaseEngineException):
     """
 
     def __init__(
-        self, instance: "Model", driver_error: pymongo.errors.DuplicateKeyError
+        self,
+        instance: "Model",
+        driver_error: Union[
+            pymongo.errors.DuplicateKeyError, pymongo.errors.BulkWriteError
+        ],
```

Comment:
I think now we only need `BulkWriteError` since we're not using update directly anymore.
```suggestion
        driver_error: pymongo.errors.BulkWriteError,
```

Reply:
Yep, makes sense! 🚀